### DISCUSSION TITLE
super() 질문

### DISCUSSION THREAD

**Discussion by ddeokbboki-good** [Boostcamp-AI-Tech-1-15/ddobokki_log#12] [3. Pytorch]
엄청나게 허접한 질문이긴 하지만, super() 함수? 이거는 언제 쓰고 언제 안 쓰는 건가요?? 
그라고 왜 괄호 안에 클래스 이름을 넣을 때도 있고 안 넣을 때도 있는데 기준이 무엇인지 잘 모르겠습니다..

**Comment by ddobokki** [comment]
super()는 부모 클래스를 호출 할 때 사용합니다.
만약 super().__init__() 이라면 부모 클래스의 __init__ 함수를 실행시키는 구문입니다.

class Parent():
  def say():
    print('부모클래스')

class Child(Parent):
  def move():
    print('걸음마')


라고 한다면
Child().say()를 호출 할 수 있습니다.(클래스 내부에서 구현을 안해도 상속 받았기 때문에 부모 클래스의 함수를 호출 할 수 있습니다.)
만약 Child 클래스 내부에 say()함수를 재정의했다면(같은 이름으로 만들었다면) 부모 클래스의 say가 아니라 자식 클래스의 say호출하구요.

**Reply by ddeokbboki-good** [reply]
아하 상속하는 거였군요. 감사합니다.
